core/domain/classifier_domain_test.py

```diff
@@ -76,10 +76,252 @@ def test_to_dict(self):
+
+        data_dict = training_job_obj.to_dict()
+        data_dict['status'] = feconf.TRAINING_JOB_STATUS_FAILED
+        with self.assertRaisesRegexp(Exception, (
+            'The status change %s to %s is not '
+            'valid.' % (
+                training_job_dict['status'],
+                data_dict['status']))):
+            training_job_obj.update_status(data_dict['status'])
+
+
+    def test_update_next_scheduled_check_time(self):
+        training_job_dict = {
+            'job_id': 'exp_id1.SOME_RANDOM_STRING',
+            'algorithm_id': 'TextClassifier',
+            'interaction_id': 'TextInput',
+            'exp_id': 'exp_id1',
+            'exp_version': 1,
+            'next_scheduled_check_time':
+                datetime.datetime.strptime(
+                    '2017-08-11 12:42:31', '%Y-%m-%d %H:%M:%S'),
+            'state_name': 'a state name',
+            'status': 'FAILED',
+            'training_data': [
+                {
+                    'answer_group_index': 1,
+                    'answers': ['a1', 'a2']
+                },
+                {
+                    'answer_group_index': 2,
+                    'answers': ['a2', 'a3']
+                }
+            ],
+            'classifier_data': {},
+            'data_schema_version': 1
+        }
+        training_job_obj = self._get_training_job_from_dict(
+            training_job_dict)
+        training_job_obj.update_next_scheduled_check_time(
+            training_job_dict['next_scheduled_check_time'])
+        data_dict = training_job_obj.to_dict()
+        self.assertTrue(data_dict['next_scheduled_check_time'])
+
+
+    def test_update_classifier_data(self):
+        training_job_dict = {
+            'job_id': 'exp_id1.SOME_RANDOM_STRING',
+            'algorithm_id': 'TextClassifier',
+            'interaction_id': 'TextInput',
+            'exp_id': 'exp_id1',
+            'exp_version': 1,
+            'next_scheduled_check_time':
+                datetime.datetime.strptime(
+                    '2017-08-11 12:42:31', '%Y-%m-%d %H:%M:%S'),
+            'state_name': 'a state name',
+            'status': 'FAILED',
+            'training_data': [
+                {
+                    'answer_group_index': 1,
+                    'answers': ['a1', 'a2']
+                },
+                {
+                    'answer_group_index': 2,
+                    'answers': ['a2', 'a3']
+                }
+            ],
+            'classifier_data': {
+                'answer_group_index': 1,
+                'answers': ['a1', 'a2']
+            },
+            'data_schema_version': 1
+        }
+        training_job_obj = self._get_training_job_from_dict(
+            training_job_dict)
+        training_job_obj.update_classifier_data(
+            training_job_dict['classifier_data'])
+        data_dict = training_job_obj.to_dict()
+        self.assertTrue(data_dict['classifier_data'])
+
+    def setup_training_job_dict(self):
```

Comment:
You do not need this. Use the `setUp` function for test. (refer [this](https://github.com/oppia/oppia/blob/develop/core/domain/activity_domain_test.py#L26))